### PR TITLE
API call for posting new data, getting all post data, like, dislike, bookmark and remove from bookmark

### PR DESCRIPTION
**Pull request by Developers-codz**
## Files to check 
`src/Redux/Reducers/postsSlice.js`
`src/components/modal/Modal.jsx`
`src/components/modal/modalComponent.js`
`src/components/posts/Post.jsx`
`src/pages/Profile/Tab/Tabs.jsx`

## [Live](https://share-arts-git-postapiintegration-developers-codz.vercel.app/)

Thank you for reviewing my PR :)

### CODE REVIEW

File: src/Redux/Reducers/postsSlice.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        })
+        .addCase(bookmark.fulfilled,(state,action) =>{
+            SuccessToast("Added to bookmark");
+            state.bookmarked = action.payload.bookmarks;
+        })
+        .addCase(bookmark.rejected,(state,action)=>{
+            AlertToast(`${action.payload.errors}`)
+        })
+    }
+})
```
Comment (b0llu) [inline src/Redux/Reducers/postsSlice.js]: remove console logs from fulfilled states after done with them

File: src/components/modal/Modal.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import {useState} from "react"
+import { AlertToast } from "../toasts";
+import { addNewPost } from "../../Redux/Reducers/postsSlice";
+
+export const Modal = ({isModalOpen,setModalOpen}) =>{
+    const dispatch = useDispatch()
+    const [postData,setPostData] = useState({content:""})
+
+    const clickHandler = () =>{
+        // console.log(postData)
```
Comment (b0llu) [inline src/components/modal/Modal.jsx]: Remove commented code

File: src/pages/Profile/Tab/Tabs.jsx [hunk 3]
```diff
@@ -32,13 +42,17 @@ export const Tabs = () => {
       </TabHead>
       <Outlet>
       <TabContent id="tab1" activeTab={activeTab}>
-         <Post />
+         {/* <Post /> */}
```
Comment (b0llu) [inline src/pages/Profile/Tab/Tabs.jsx:45]: Remove commented code

**Review by b0llu — APPROVED** [review]
Suggested few things. Everything else looks good you can merge this PR

File: src/Redux/Reducers/postsSlice.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        return response.data;
+    }
+    catch(error){
+        return rejectWithValue(error.response.data)
+    }
+})
+
+export const addNewPost = createAsyncThunk("posts/addNewPost",async (post,{rejectWithValue})=>{
+    const encodedToken = localStorage.getItem("token")
+    console.log(post)
```
Comment (Maianki) [inline src/Redux/Reducers/postsSlice.js]: You can remove this log statement

File: src/Redux/Reducers/postsSlice.js (hunk 1 above)
Comment (Developers-codz) [reply src/Redux/Reducers/postsSlice.js]: Okay, Done

File: src/pages/Profile/Tab/Tabs.jsx (hunk 3 above)
Comment (Developers-codz) [reply src/pages/Profile/Tab/Tabs.jsx:45]: Done

**Review by Maianki — APPROVED** [review]
Looks good you can go ahead with merge.

File: src/Redux/Reducers/postsSlice.js (hunk 4 above)
Comment (Developers-codz) [reply src/Redux/Reducers/postsSlice.js]: Done

**Pull request merged**